libs/blocks/global-navigation/global-navigation.js

```diff
@@ -710,21 +714,23 @@ class Gnav {
   };
 }
 
-export default async function init(header) {
-  const { locale } = getConfig();
-  // TODO locale.contentRoot is not the fallback we want if we implement centralized content
-  const url = getMetadata('gnav-source') || `${locale.contentRoot}/gnav`;
-  const resp = await fetch(`${url}.plain.html`);
-  const html = await resp.text();
-  if (!html) return null;
-  const parsedHTML = await replaceText(html, getFedsPlaceholderConfig(), undefined, 'feds');
-  processMartechAttributeMetadata(parsedHTML);
-
+export default async function init(block) {
   try {
-    const gnav = new Gnav(new DOMParser().parseFromString(parsedHTML, 'text/html').body, header);
+    const { locale } = getConfig();
+    const url = getMetadata('gnav-source') || `${locale.contentRoot}/gnav`;
+    const content = await fetchAndProcess({
+      url,
+      e: 'Error fetching gnav content',
```

Comment:
I think `e` should be `message`? I added some thoughts on passing this in the utilities file

Reply:
Slightly refactored this part